#**Quest: Shark Attacks**

##**General Information on dataset**

All individuals survived unless noted otherwise.
Entries on the spreadsheet are color-coded.

• Unprovoked Incidents = Tan
• Provoked Incidents = Orange
• Incidents Involving Watercrafts = Green
• Air / Sea Disasters = Yellow
• Questionable Incidents = Blue

Unprovoked vs. Provoked - GSAF defines a provoked incident as one in which the shark was speared, hooked, captured or in which a human drew "first blood". Although such incidents are of little interest to shark behaviorists, when the species of shark involved is known and pre-op photos of the wounds are available, the bite patterns are of value in determining species of shark involved in other cases when the species could not identified by the patient or witnesses. We know that a live human is rarely perceived as prey by a shark. Many incidents are motivated by curiosity, others may result when a shark perceives a human as a threat or competitor for a food source, and could be classed as "provoked" when examined from the shark's perspective.

Incidents involving watercraft – Incidents in which a boat was bitten or rammed by a shark are in green. However, in cases in which the shark was hooked, netted or gaffed, the entry is orange because they are classed as provoked incidents.

Questionable incidents - Incidents in which there are insufficient data to determine if the injury was caused by a shark or the person drowned and the body was later scavenged by sharks. In a few cases, despite media reports to the contrary, evidence indicated there was no shark involvement whatsoever. Such incidents are in blue.

source: https://www.sharkattackfile.net/incidentlog.htm

**DAY 1: CHECKLIST

In [2]:
#1. Downloaded the Shark Attack dataset and successfully imported it into your 
#Python environment.

import pandas as pd
file_location= "GSAF5.xls"
shark_df=pd.read_excel(file_location)

In [32]:
#2. Performed an initial analysis of the dataset to understand its structure and what the 
#ields represent.
shark_df.head() #Quick pick on how the DF looks like

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,...,Species,Source,pdf,href formula,href,Case Number,Case Number.1,original order,Unnamed: 21,Unnamed: 22
0,14/04,2026.0,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,...,Unknown,The U.S. Sun: Simon De Marchi,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,3rd April,2026.0,Unprovoked,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,Oliver Tokic-Bensley,M,16,...,Bronze Whaler,ABC News: The Guardian: Andrew Currie and Bob...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,26th March,2026.0,Unprovoked,Bahamas,Andros Island,Fresh Creek,Swimming,Australian woman,F,22,...,Unknown,WIC News: Melissa Michaelson,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,25th March,2026.0,Unprovoked,Australia,South Australia,Cape Jaffa Limestone Coast,Diving,Luke Kuhn,M,?,...,Great White Shark 3.5m (11.5ft),Bob Myatt GSAF,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,22nd-23rd March,2026.0,Unprovoked,Australia,NSW,Little Avalon Beach,Surfing,Unknown,M,30+,...,Unknown,Melissa Michaelson: Instagram,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [23]:
print(f"# rows: {shark_df.shape[0]} and # columns:  {shark_df.shape[1]}")
print("=======================")
print(shark_df.info())

# rows: 7087 and # columns:  23
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7087 entries, 0 to 7086
Data columns (total 23 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Date            7087 non-null   object 
 1   Year            7085 non-null   float64
 2   Type            7069 non-null   object 
 3   Country         7037 non-null   object 
 4   State           6600 non-null   object 
 5   Location        6520 non-null   object 
 6   Activity        6504 non-null   object 
 7   Name            6869 non-null   object 
 8   Sex             6509 non-null   object 
 9   Age             4093 non-null   object 
 10  Injury          7051 non-null   object 
 11  Fatal Y/N       6526 non-null   object 
 12  Time            3560 non-null   object 
 13  Species         3956 non-null   object 
 14  Source          7067 non-null   object 
 15  pdf             6799 non-null   object 
 16  href formula    6794 non-null   object 
 17  h

#**Data Frame Analysis:**
##Issues found:

* Analysis per column:
Date: has different formats, and it is string type. Recommendation: unify the format using only numbers e.g. DD.MM.YY
Age: has different formats, and it is string type. Recommendation: unify the format using only integer numbers
pdf: seems useles, check whether is always NaN or there is somethig relevant like links to files
href formula: unclear what this means
href: unclear what this means 
Case Number: many empty fields, to be checked in detail
Case Number.1: many empty fields, to be checked in detail, check if it means the same as "Case Number" column
original order: unclear what this means 
Unnamed: unclear what this means, check if holds any value other than NaN, otherwise delete column 
Unnamed: unclear what this means, check if holds any value other than NaN, otherwise delete column

* Not all columns have the same number of rows, important to determine the maximum amount of rows to map only the relevant information. 


##Optional improvements:
Year: change the type to integer (currently float)

In [28]:
#Analyzing the description for non-numerical columns
shark_df.describe(include="object") 

,Date,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,...,Time,Species,Source,pdf,href formula,href,Case Number,Case Number.1,Unnamed: 21,Unnamed: 22
count,7087,7069,7037,6600,6520,6504,6869,6509,4093,7051,...,3560,3956,7067,6799,6794,6796,6798,6797,1,2
unique,6126,14,253,948,4631,1612,5801,11,252,4197,...,477,1746,5413,6789,6784,6776,6777,6775,1,2
top,1957,Unprovoked,USA,Florida,"New Smyrna Beach, Volusia County",Surfing,male,M,16,FATAL,...,Afternoon,White shark,"K. McMurray, TrackingSharks.com",1907.10.16.R-HongKong.pdf,http://sharkattackfile.net/spreadsheets/pdf_di...,http://sharkattackfile.net/spreadsheets/pdf_di...,2021.07.23,2012.09.02.b,stopped here,Teramo
freq,9,5237,2580,1192,191,1151,679,5682,92,863,...,215,194,131,2,2,4,2,2,1,1


###Individual column analysis:
* Column "Type": the entries shall be unified, there are typo errors creating new types. This column seems to be aligned with the color coding explain in the introduction of the shark attack log file, from which there should be only the following types: 
• Unprovoked Incidents = Tan
• Provoked Incidents = Orange
• Incidents Involving Watercrafts = Green
• Air / Sea Disasters = Yellow
• Questionable Incidents = Blue

*Dataframe has 7069 types, while color coding only 5 or 6, o mirar si Unprovoked and Provoked only and the other sub-types.

* Columns "Country", "State", "Location", "Activity": the entries shall be unified, there might be typo errors creating too many entries.

* Column "Sex": reduce (potentially 2: F,M, check if other exists), 11 seems not realistic.

* Column "Time": check if needed, if it is, could be categorized to unify entries.